aten/src/ATen/native/transformers/cuda/attention.cu

```diff
@@ -875,18 +883,40 @@ std::tuple<at::Tensor, at::Tensor> _efficient_attention_forward(
 
   at::Tensor res;
   at::Tensor logsumexp;
+  at::Tensor seed_t, offset_t;
 
   const bool use_dropout = std::fpclassify(dropout_p) != FP_ZERO;
-  at::PhiloxCudaState rng_engine_inputs;
+  at::PhiloxCudaState philox_state;
   if (use_dropout) {
-    at::CUDAGeneratorImpl* gen =
-        at::get_generator_or_default<at::CUDAGeneratorImpl>(
-            c10::nullopt, at::cuda::detail::getDefaultCUDAGenerator());
+    auto gen = at::get_generator_or_default<at::CUDAGeneratorImpl>(
+        c10::nullopt, at::cuda::detail::getDefaultCUDAGenerator());
 
+    // See Note [Acquire lock when using random generators]
     std::lock_guard<std::mutex> lock(gen->mutex_);
     // if using dropout, we produce 1 random number for each element of the
     // attention tensor
-    rng_engine_inputs = gen->philox_cuda_state(B * num_heads * M * N);
+    philox_state = gen->philox_cuda_state(B * num_heads * M * N);
+
+    if (at::cuda::currentStreamCaptureStatus() ==
+        at::cuda::CaptureStatus::None) {
+      auto [seed, offset] = at::cuda::philox::unpack(philox_state);
```

Comment:
nit: I dislike auto for very low level, critical work. I'd rather this fail compilation than for the upcoming cast to succeed and introduce silent correctness.